### PR TITLE
v1.4 PED fix & Update of PPP [Complete]

### PR DESCRIPTION
**Pull request by DonovanJJ**
## The following PED bug fixes were made:
- 91, 93, 122 (Removed the existing add command)
- 127, 128, 131 (Convert multiple white spaces into single white spaces and trimmed the user input)
- 104 (UG `remove /c` command was updated)
- 132 (Add checking of attendance boundary when marking student present)
- 106 (Update the error message for remove such that the correct  message is shown, previously was a bug)

## The following bug was fixed (Non-PED):
- 81 (Cascaded the removal of class to students as well)

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -871,10 +870,10 @@ There is no need to save manually.
...
 | **view**        | **Viewing a class:** <br> **Format:** `view /c CLASS_INDEX` <br> **E.g:** `view /c 1`                                                                                                                                                                                                                                                                    |
-| **edit**        | **Editing a class:** <br> **Format:** `edit /c CLASS_INDEX [/n CLASS_NAME] [/m MEMO] [/t SCHEDULE]` <br> **E.g:** `edit /c 2 /m submit marking report` <br><br> **Editing a student:** <br> **Format:** `edit /s STUDENT_INDEX /c CLASS_NAME [/n STUDENT_NAME] [/id ID] [/m MEMO] [/p PARTICIPATION]` <br> **E.g:** `edit /s 1 /c cs2103t /id A1234567W` |
+| **edit**        | **Editing a class:** <br> **Format:** `edit /c CLASS_INDEX [/n CLASS_NAME] [/m MEMO] [/t SCHEDULE]` <br> **E.g:** `edit /c 2 /m submit marking report` <br><br> **Editing a student:** <br> **Format:** `edit /s STUDENT_INDEX /c CLASS_NAM  E [/n STUDENT_NAME] [/id ID] [/m MEMO] [/p PARTICIPATION]` <br> **E.g:** `edit /s 1 /c cs2103t /id A1234567W` |
```
Comment (songgthu) [inline docs/UserGuide.md]: just a small typo here, there's an extra space in "CLASS_NAME"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -871,10 +870,10 @@ There is no need to save manually.
...
-| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME /c CLASS_INDEX` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
-| **remove**      | **Removing a class:** <br> **Format:** `remove /c CLASS_INDEX` <br> **E.g:** `remove /c 1` <br> <br> **Removing a student:** <br> **Format:** `remove /s STUDENT_INDEX  /c CLASS_NAME` <br> **E.g:** `remove /s 1 /c CS2103T`                                                                                                                            |
+| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME  /c CLASS_NAME` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
```
Comment (songgthu) [inline docs/UserGuide.md]: I think for the adding a student the format is add /s STUDENT_NAME /c CLASS_INDEX, can you update it and the following example also thanks!

**Review by songgthu — APPROVED** [review]
LGTM! There are some small details to be updated.

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 3]
```diff
@@ -145,41 +136,41 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
...
+    //        logic = new LogicManager(model, storage);
+    //
+    //        // Triggers the saveAddressBook method by executing an add command
+    //        String addCommand = AddCommand.COMMAND_WORD + NAME_DESC_AMY + PHONE_DESC_AMY
+    //                + EMAIL_DESC_AMY + ADDRESS_DESC_AMY;
+    //        Student expectedPerson = new StudentBuilder(AMY).build();
+    //        ModelManager expectedModel = new ModelManager();
+    //        expectedModel.addStudent(expectedPerson);
+    //        assertCommandFailure(addCommand, CommandException.class, expectedMessage, expectedModel);
+    //    }
```
Comment (sudarshan2401) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: Is this part of the test not going to be used? If so maybe can remove

**Review by sudarshan2401 — APPROVED** [review]
LGTM! Thanks for the bug fixes!

File: docs/UserGuide.md (hunk 1 above)
Comment (DonovanJJ) [reply docs/UserGuide.md]: Thanks for spotting this mistake, I've made the necessary changes to this and the one you pointed out in the command summary!

File: src/test/java/seedu/address/logic/LogicManagerTest.java (hunk 3 above)
Comment (DonovanJJ) [reply src/test/java/seedu/address/logic/LogicManagerTest.java]: Thanks for pointing it out! I've removed this test

**Pull request merged**